libs/libxx/0001-libcxx-enable-support-for-c++11.patch

```diff
@@ -0,0 +1,49 @@
+diff --git a/libcxx/src/barrier.cpp b/libcxx/src/barrier.cpp
+index 54890b2d797e..13bc65dfb307 100644
+--- a/libcxx/src/barrier.cpp
++++ b/libcxx/src/barrier.cpp
```

Comment:
why need to support the old compiler if the official repo drop them.

Reply:
As I understand, LLVM still has support for C++11. From `libcxx/include/__config`:
```c++
#ifndef _LIBCPP_STD_VER
#  if  __cplusplus <= 201103L
#    define _LIBCPP_STD_VER 11
#  elif __cplusplus <= 201402L
#    define _LIBCPP_STD_VER 14
#  elif __cplusplus <= 201703L
#    define _LIBCPP_STD_VER 17
#  elif __cplusplus <= 202002L
#    define _LIBCPP_STD_VER 20
#  else
#    define _LIBCPP_STD_VER 21  // current year, or date of c++2b ratification
#  endif
#endif // _LIBCPP_STD_VER
```
and
```c++
#if defined(_LIBCPP_COMPILER_GCC) && __cplusplus < 201103L
#error "libc++ does not support using GCC with C++03. Please enable C++11"
#endif
```
I would say that their minimum supported version is C++11 (with GCC).

Was the intention to drop C++11 from NuttX, when using the LLVM library?

Reply:
As far as I know, the user could select C++ 11, but the build of libcxx need at least c++17: https://github.com/apache/nuttx/pull/6181.
Since NuttX build libcxx and user code in one pass with the same toolchain in most user case, it doesn't make sense to use the different c++ standard.

Reply:
Ok. Sure, that makes sense. I'll remove that patch. Would it be useful to also add additional text the the help prompt of `CONFIG_LIBCXX` to state that it requires c++17 or later?

Reply:
Yes, we definitely need more documentation.